src/pages/DirectMessage.tsx

```diff
@@ -0,0 +1,252 @@
+import {
+  Avatar,
+  Box,
+  Divider,
+  InputAdornment,
+  List,
+  ListItem,
+  ListItemText,
+  Stack,
+  TextField,
+  Typography,
+} from "@mui/material";
+import IconButton from "@mui/material/IconButton";
+import { useEffect, useRef, useState } from "react";
+import axios from "axios";
+import { useParams } from "react-router-dom";
+import { useAppDispatch, useAppSelector } from "../state/hooks";
+import InfoOutlinedIcon from "@mui/icons-material/InfoOutlined";
+import AddPhotoAlternateOutlinedIcon from "@mui/icons-material/AddPhotoAlternateOutlined";
+import EmojiEmotionsOutlinedIcon from "@mui/icons-material/EmojiEmotionsOutlined";
+import GifBoxOutlinedIcon from "@mui/icons-material/GifBoxOutlined";
+import SendIcon from "@mui/icons-material/Send";
+import ConversationList from "../components/Messages/ConversationList";
+import {
+  setSelectedConversation,
+  updateConversation,
+} from "../state/slices/messagesSlice";
+
+const styles = {
+  directMessageActivityContainer: {
+    width: "50%",
+    display: "flex",
+    flexDirection: "column",
+    height: "100%",
+  },
+  directMessageActivityHeader: {
+    display: "flex",
+    justifyContent: "space-between",
+    paddingLeft: 2,
+    paddingRight: 2,
+  },
+  message: {
+    display: "flex",
+    flexDirection: "column",
+    alignItems: "flex-start",
+  },
+  messageText: {
+    padding: 1,
+    borderRadius: 10,
+    backgroundColor: "#cce3d9",
+  },
+  root: {
+    width: "100%",
+  },
+  sentMessage: {
+    display: "flex",
+    flexDirection: "column",
+    alignItems: "flex-end",
+  },
+  sentMessageText: {
+    padding: 1,
+    borderRadius: 10,
+    backgroundColor: "#22AA6F",
+  },
+};
+
+export type Message = {
+  messageId: number;
+  timestamp: string;
+  textContent: string;
+  sentUserId: number;
+  receivedUserId: number;
+};
+
+const DirectMessage = () => {
+  const { userId1, userId2 } = useParams();
+  const [textContent, setTextContent] = useState("");
+  const [messages, setMessages] = useState<Message[]>([]);
+  const user = useAppSelector((state) => state.user);
+  const { selectedConversation } = useAppSelector((state) => state.messages);
+  const messageRef = useRef<HTMLDivElement>(null);
+  const dispatch = useAppDispatch();
+
+  useEffect(() => {
+    const fetchDirectMessage = async () => {
+      const result = await axios.get(
+        `http://localhost:3001/api/messages/${userId1}/${userId2}`
+      );
+      setMessages(result.data.messages as Message[]);
+      dispatch(
+        setSelectedConversation({
+          ...result.data.otherUser,
+          userId: Number(userId2),
+        })
+      );
+    };
+    fetchDirectMessage();
+  }, [dispatch, userId1, userId2]);
+
+  useEffect(() => {
+    // TODO: See if this is the best way to scroll to the bottom, and check edge cases
+    if (messageRef.current) {
+      messageRef.current.scrollTo(0, messageRef.current.scrollHeight);
```

Comment:
since this is a conditional, let's remove the if-statement and make it something like: 

```suggestion
      messageRef.current?.scrollTo(0, messageRef.current.scrollHeight);
```

Reply:
Yeah, nice suggestion